transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -979,4 +974,22 @@ public void stop() {
     }
   }
 
+  private static class XAEvictionVeto<K, V> implements EvictionVeto<K, SoftLock<V>> {
+
+    private final EvictionVeto<? super K, ? super V> wrappedVeto;
+
+    private XAEvictionVeto(EvictionVeto<? super K, ? super V> wrappedVeto) {
+      this.wrappedVeto = wrappedVeto;
+    }
+
+    @Override
+    public boolean vetoes(K key, SoftLock<V> softLock) {
+      if (isInDoubt(softLock)) {
```

Comment:
My IDE agrees with you ... still find the expanded version more readable.
Others to chime in?